tgui/packages/tgui/interfaces/SDQLSpellMenu.js

```diff
@@ -0,0 +1,476 @@
+  const { data } = useBackend(context);
+  const {
+    type,
+  } = data;
+
+  const vars = typevars(type);
+
+  return (
+    <Section fill scrollable>
+      {vars.filter(entry => var_condition(entry, context)).map(entry => (
+        <Stack key={entry.name} mb="6px">
+          <Stack.Item>
+            <WrapInTooltip entry={entry}>
+              <Box inline bold color="label" mr="6px">
+                {entry.name}:
+              </Box>
+            </WrapInTooltip>
+          </Stack.Item>
+          <Stack.Item shrink basis="100%">
+            <SDQLSpellOption entry={entry} />
+          </Stack.Item>
+        </Stack>
+      ))}
+    </Section>
+  );
+};
+
+const SDQLSpellOption = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    saved_vars,
+  } = data;
+  const {
+    entry,
+  } = props;
+  switch (entry.type) {
+    case "string":
+      return (
+        <Input
+          width="100%"
+          fluid
+          value={(Object.prototype.hasOwnProperty.call(saved_vars, entry.name) 
+            && saved_vars[entry.name]) ?? entry.default}
+          onChange={(e, value) => act('variable', { name: entry.name, value: value })} />
+      );
+    case "int":
+      return (
+        <NumberInput
+          value={(Object.prototype.hasOwnProperty.call(saved_vars, entry.name) 
+            && saved_vars[entry.name]) ?? entry.default}
+          onChange={(e, value) => act('variable', { name: entry.name, value: value })} />
+      );
+    case "bool":
+      return (
+        <Button.Checkbox
+          checked={
+            (Object.prototype.hasOwnProperty.call(saved_vars, entry.name) 
+            && saved_vars[entry.name]) ?? entry.default
+          }
+          onClick={() => act('bool_variable', { name: entry.name })} />
+      );
+    case "string_enum":
+      return (
+        <Dropdown
+          options={entry.options}
+          displayText={
+            (Object.prototype.hasOwnProperty.call(saved_vars, entry.name) 
+            && saved_vars[entry.name]) ?? entry.default
+          }
+          onSelected={value => act('variable', { name: entry.name, value: value })} />
+      );
+    case "int_enum":
+      return (
+        <Dropdown
+          options={entry.options}
+          displayText={entry.options[
+            (Object.prototype.hasOwnProperty.call(saved_vars, entry.name) 
+            && saved_vars[entry.name])] ?? entry.default}
+          onSelected={value => act('variable', { name: entry.name, value: entry.options.indexOf(value) })} />
+      );
+    case "list":
+      return (
+        <SDQLSpellListEntry list={entry.name} />
+      );
+  }
+};
+
+const SDQLSpellListEntry = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    list_vars,
+  } = data;
+  const {
+    list,
+  } = props;
+  return (
+    <Collapsible>
+      {Object.prototype.hasOwnProperty.call(list_vars, list) 
+        && Object.entries(list_vars[list]).map(entry => (
```

Comment:
............
```
list_vars?.[list]
```

Please destructure your entries in the arguments, with clear names, instead of `entry[0]`, `entry[1]`
```
([key, value]) => {
```